core/src/main/java/org/apache/carbondata/core/memory/UnsafeMemoryManager.java

```diff
@@ -124,8 +116,21 @@ private synchronized MemoryBlock allocateMemory(MemoryType memoryType, String ta
             + "Bytes");
       }
       return allocate;
+    } else {
+      MemoryBlock allocate = MemoryAllocator.HEAP.allocate(memoryRequested);
```

Comment:
Add a INFO logger for fallback case from off  heap to on-heap using the below condition
`if (memoryUsed + memoryRequested <= totalMemory && memoryType == MemoryType.OFFHEAP) {
    Logger.info("Configured unsafe memory is not sufficient so falling back to unsafe on-heap");
}`

Reply:
Its not actually fall back once offheap memory is available it will switch to offheap ...depends on offheap memory , i will add debug log